File: delivery/php-sdk/models/sender/api/delivery_change.php
<?php
/**
 * @Author Hanqiang
 * @Date 2025/12/9
 * @PackageName: models/sender/api
 * @ClassName: delivery_change
 * @Description: 麦芽田配送开放平台SDK - 配送状态同步接口
 * 提供同步订单配送状态操作的功能
 * @Version 1.0
 */

namespace Maiyatian\Delivery\SDK\models\sender\api;

use Maiyatian\Delivery\SDK\client\HTTPClientManager;
use Maiyatian\Delivery\SDK\models\types\Constants;

class DeliveryChange extends Base
{
    /**
     * 同步订单配送状态
     *
     * @param string $orderNo 配送单号
     * @param string $sourceOrderNo 三方订单号
     * @param string $shopId 三方商户ID、门店ID
     * @param string $status 配送状态
     * @param string $riderName 骑手名称
     * @param string $riderPhone 骑手电话
     * @param string $longitude 经度
     * @param string $latitude 纬度
     * @param string $pickupCode 取货码
     * @param int $distance 配送距离（单位：米）
     * @param int $deliveryFee 配送费（单位：分）
     * @param int $cancelType 取消类型
     * @param string $cancelReason 取消原因
     * @param int $cancelDeditAmount 取消订单违约金（单位：分）
     * @param array $vehicleInfo 车辆信息
     * @param int $updateTime 更新时间
     * @param bool $isTransship 是否接驳单
     * @return HTTPClientManager\HTTPResponse
     */
    public function syncDeliveryStatus(
        string $orderNo,
        string $sourceOrderNo,
        string $shopId,
        string $status,
        string $riderName = '',
        string $riderPhone = '',
        string $longitude = '',
        string $latitude = '',
        string $pickupCode = '',
        int $distance = 0,
        int $deliveryFee = 0,
        int $cancelType = 0,
        string $cancelReason = '',
        int $cancelDeditAmount = 0,
        array $vehicleInfo = [],
        int $updateTime = 0,
        bool $isTransship = false
    ): HTTPClientManager\HTTPResponse {
        if ($updateTime === 0) {
            $updateTime = time();
        }

        $data = [
            'order_no' => $orderNo,
            'source_order_no' => $sourceOrderNo,
            'shop_id' => $shopId,
            'status' => $status,
            'rider_name' => $riderName,
            'rider_phone' => $riderPhone,
            'longitude' => $longitude,
            'latitude' => $latitude,
            'pickup_code' => $pickupCode,
            'distance' => $distance,
            'delivery_fee' => $deliveryFee,
            'cancel_type' => $cancelType,
            'cancel_reason' => $cancelReason,
            'cancel_dedit_amount' => $cancelDeditAmount,
            'vehicle_info' => $vehicleInfo,
            'update_time' => $updateTime,
            'is_transship' => $isTransship,
        ];

        return $this->post(Constants::COMMAND_DELIVERY_CHANGE, $data);
    }
}

File: delivery/go-sdk/models/sender/api/delivery_change.go
/**
 * @Author Hanqiang
 * @Date 2025/12/9
 * @PackageName: api
 * @ClassName: delivery_change
 * @Description: 配送状态同步API
 * @Version 1.0
 */

package api

import (
	"context"
	"net/http"

	"github.com/anthonyhann/myt-open-api-sdk/delivery/go-sdk/client"
	entity "github.com/anthonyhann/myt-open-api-sdk/delivery/go-sdk/models/sender/entity/delivery"
)

const (
	// DeliveryChange 配送状态同步接口路径
	// command: delivery_change
	// 必接: 是
	DeliveryChange = "v1/delivery/delivery_change"
)

// DeliveryChange 配送状态同步
//
// 功能说明:
//   - 同步订单配送状态操作
//   - 三方配送服务商主动推送订单配送状态到麦芽田平台
//   - 支持配送全流程状态更新（接单、取货、配送中、完成、取消等）
//
// 参数说明:
//   - ctx: 上下文对象，用于控制请求的超时和取消
//   - token: 麦芽田授权token，用于身份验证
//   - data: 请求参数，包含订单号、状态、骑手信息、位置等
//
// 返回值:
//   - *client.HTTPResponse: 标准响应，code为200表示成功
//   - error: 错误信息，成功时为nil
//
// 状态说明:
//   - GRABBED: 已分配骑手
//   - PICKUP: 待取货
//   - DELIVERING: 配送中
//   - DONE: 已完成
//   - CANCEL: 已取消
//   - RIDER_CANCEL: 骑手取消
//   - TRANSFER: 骑手转单
//   - EXPECT: 配送异常
//
// 注意事项:
//   - 订单状态变更时必须及时推送
//   - 取消状态时需提供cancel_type和cancel_reason
//   - 骑手信息（姓名、电话、位置）建议实时更新
//
// 使用示例:
//
//	req := &entity.DeliveryChangeReq{
//	    OrderNo:       "delivery_order_123",
//	    SourceOrderNo: "myt_order_456",
//	    ShopId:        "shop_789",
//	    Status:        "DELIVERING",
//	    RiderName:     "张骑手",
//	    RiderPhone:    "13800138000",
//	    Longitude:     "116.397128",
//	    Latitude:      "39.916527",
//	    UpdateTime:    time.Now().Unix(),
//	}
//	resp, err := sender.DeliveryChange(ctx, token, req)
//	if err != nil {
//	    log.Fatal(err)
//	}
func (o *deliverySender) DeliveryChange(ctx context.Context, token string, data *entity.DeliveryChangeReq) (*client.ApiResponse[ApiResponseEmptyData], error) {
	return client.RequestWithApiClient[ApiResponseEmptyData](ctx, o.apiClient, http.MethodPost, DeliveryChange, data, token, nil)
}
